src/services/rsvp/rsvp-router.ts

```diff
@@ -0,0 +1,181 @@
+import { Request, Response, Router } from "express";
+import { StatusCode } from "status-code-enum";
+import { strongJwtVerification } from "../../middleware/verify-jwt.js";
+import { JwtPayload } from "../auth/auth-models.js";
+import { hasElevatedPerms } from "../auth/auth-lib.js";
+import { DecisionStatus, DecisionResponse, DecisionInfo } from "../../database/decision-db.js";
+import Models from "../../database/models.js";
+
+const rsvpRouter: Router = Router();
+
+/**
+ * @api {get} /rsvp/:USERID/ GET /rsvp/:USERID/
+ * @apiGroup rsvp
+ * @apiDescription Check RSVP decision for a given userId, provided that the current user has elevated perms
+ *
+ *
+ * @apiSuccess (200: Success) {string} MongoDB object ID
+ * @apiSuccess (200: Success) {string} userId
+ * @apiSuccess (200: Success) {string} User's applicatoin status
+ * @apiSuccess (200: Success) {string} User's Response (whether or whether not they're attending)
+ * @apiSuccess (200: Success) {string} Reviwer
+ * @apiSuccess (200: Success) {boolean} Whether email has been sent
+ * @apiSuccessExample Example Success Response:
+ * 	HTTP/1.1 200 OK
+ *	{
+ *      "_id": "652c311b6e283244d2ef4c29",
+ *      "userId": "github0000001",
+ *      "status": "ACCEPTED",
+ *      "response": "PENDING",
+ *      "reviewer": "reviewer1",
+ *      "emailSent": true
+ * 	}
+ *
+ * @apiUse strongVerifyErrors
+ */
+rsvpRouter.get("/:USERID", strongJwtVerification, async (req: Request, res: Response) => {
+    const userId: string | undefined = req.params.USERID;
+
+    const payload: JwtPayload = res.locals.payload as JwtPayload;
+    //Redirects if caller doesn't have elevated perms
+    if (!hasElevatedPerms(payload)) {
+        return res.redirect("/");
+    }
+
+    const queryResult: DecisionInfo | null = await Models.DecisionInfo.findOne({ userId: userId });
+
+    //Returns error if query is empty
+    if (!queryResult) {
+        return res.status(StatusCode.ClientErrorBadRequest).send({ error: "UserNotFound" });
+    }
+
+    return res.status(StatusCode.SuccessOK).send(queryResult.toObject());
+});
+
+/**
+ * @api {get} /rsvp/ GET /rsvp/
+ * @apiGroup rsvp
+ * @apiDescription Check RSVP decision for current user, returns filtered info for attendees and unfiltered info for staff/admin
+ *
+ *
+ * @apiSuccess (200: Success) {string} MongoDB object ID
+ * @apiSuccess (200: Success) {string} userId
+ * @apiSuccess (200: Success) {string} User's applicatoin status
+ * @apiSuccess (200: Success) {string} User's Response (whether or whether not they're attending)
+ * @apiSuccessExample Example Success Response (caller is a user):
+ * 	HTTP/1.1 200 OK
+ *	{
+ *      "_id": "652c311b6e283244d2ef4c29",
```

Comment:
_id shouldn't be mentioned in the docs, this is a confusing thing to include and long term we hope to remove it entirely